pistache 0.4.26 (new formula)

dgreatwood opened the pull request:
Adding Pistache to homebrew-core for the first time.

Pistache is a modern, fast, elegant HTTP + REST C++17 framework with a pleasant API. It supports Linux (including via Debian - https://packages.debian.org/sid/libpistache-dev), macOS, Windows and BSD.

<!-- Use [x] to mark item done, or just click the checkboxes with device pointer -->

- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

dgreatwood commented:
Looks like there is an error in linuxbrew case, missing libcurl dependency, I'll address and retest. Please standby for now. Thx.

chenrui333 commented on Formula/p/pistache.rb:
```diff
@@ -0,0 +1,161 @@
+#
+# SPDX-FileCopyrightText: 2024 Duncan Greatwood
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+
```
```suggestion
```

dgreatwood replied:
Done

chenrui333 commented:
> Looks like there is an error in linuxbrew case, missing libcurl dependency, I'll address and retest. Please standby for now. Thx.

you can add `uses_from_macos "curl"`

SMillerDev commented on Formula/p/pistache.rb:
```diff
@@ -0,0 +1,161 @@
+
+  def install
+    system "meson", "setup", "build",
+           "--buildtype=release",
```
I believe this is part of the standard arguments already

dgreatwood replied:
Per the cookbook, that's correct, thanks. I removed the redundant
`--buildtype=release`

dgreatwood commented:
Now passes homebrew CI on Linux and macOS. Made a number of updates as follows:

- As suggested by @chenrui333, added libcurl as a dependency with "uses_from_macos"
- As suggested by @chenrui333, removed SPDX copyright+license message
- As suggested by @SMillerDev, removed redundant "--buildtype=release"
- Updated formula to Pistache release 0.4.26 to support "--wrap-mode=nofallback" option in Linux (was already supported for macOS in Pistache 0.4.25)
- Corrected formula to specify whether each dependency is/is not build only

chenrui333 commented on Formula/p/pistache.rb:
```diff
@@ -0,0 +1,157 @@
+  depends_on "rapidjson"
+  depends_on "zstd"
+
+  uses_from_macos "curl" => :build # For build on Linux; macOS ignores
```
needs zlib for linux build (also removed the comment, as `uses_from_macos` already covered the semantics)

```suggestion
  uses_from_macos "curl" => :build
  uses_from_macos "zlib"
```

dgreatwood replied:
OK, changed

chenrui333 commented on Formula/p/pistache.rb:
```diff
@@ -0,0 +1,157 @@
+  depends_on "ninja" => :build
+  depends_on "pkgconf" => :build
+  depends_on "brotli"
+  depends_on "howard-hinnant-date"
```
this should be build dep?
```suggestion
  depends_on "howard-hinnant-date" => :build
```

chenrui333 replied:
any thoughts about this?

dgreatwood replied:
Yes, you're correct.

Now, for clarity, Pistache's _src/common/http_defs.cc_ has:
`#include <date/date.h>`
which comes from howard-hinnant-date. _http_defs.cc_ is part of the core Pistache library, not solely build or unit-test.
However, howard-hinnant-date is a header-only library - so once the build is done, the howard-hinnant-date library (i.e. header file) is no longer needed.

Changed.

chenrui333 commented on Formula/p/pistache.rb:
```diff
@@ -0,0 +1,157 @@
+  depends_on "howard-hinnant-date"
+  depends_on "libevent"
+  depends_on "openssl@3"
+  depends_on "rapidjson"
```
this should be also build dep?
```suggestion
  depends_on "rapidjson" => :build
```

dgreatwood replied:
Yes, I think you're correct, rapidjson is for the internal Pistache unit tests.
Changed.

chenrui333 commented on Formula/p/pistache.rb:
```diff
@@ -0,0 +1,157 @@
+  license "Apache-2.0"
+  head "https://github.com/pistacheio/pistache.git", branch: "master"
+
+  depends_on "cmake" => :build # Required so meson finds hinnant-date
```
```suggestion
  depends_on "cmake" => :build # for howard-hinnant-date
```

dgreatwood replied:
OK, changed.

chenrui333 commented:
thanks for the work, left couple of comments. 👍

dgreatwood commented:
@chenrui333, thanks for your review. As you've seen, additional changes made per your comments:
- Changed / removed formula-file comments as suggested
- Made rapidjson and howard-hinnant-date into build-only dependencies
- Added zlib as a Linux dependency with uses_from_macos

carlocab commented on Formula/p/pistache.rb:
```diff
@@ -0,0 +1,158 @@
+  uses_from_macos "zlib"
+
+  def install
+    system "meson", "setup", "build",
+           "-DPISTACHE_USE_SSL=true", "-DPISTACHE_BUILD_EXAMPLES=true",
+           "-DPISTACHE_BUILD_TESTS=true",
+           "-DPISTACHE_BUILD_DOCS=false",
+           "-DPISTACHE_USE_CONTENT_ENCODING_DEFLATE=true",
+           "-DPISTACHE_USE_CONTENT_ENCODING_BROTLI=true",
+           "-DPISTACHE_USE_CONTENT_ENCODING_ZSTD=true",
+           *std_meson_args
```
```suggestion
    system "meson", "setup", "build",
                    "-DPISTACHE_USE_SSL=true",
                    "-DPISTACHE_BUILD_EXAMPLES=false",
                    "-DPISTACHE_BUILD_TESTS=false",
                    "-DPISTACHE_BUILD_DOCS=false",
                    "-DPISTACHE_USE_CONTENT_ENCODING_DEFLATE=true",
                    "-DPISTACHE_USE_CONTENT_ENCODING_BROTLI=true",
                    "-DPISTACHE_USE_CONTENT_ENCODING_ZSTD=true",
                    *std_meson_args
```

1. Follow the style used by other formulae (see, e.g., `libgusb` or `libjcat`)
2. Skip building the tests or examples since it doesn't look like we're using/running them anywhere

dgreatwood replied:
@carlocab and @chenrui333 - changed and pushed again

carlocab reviewed (approved):
Thanks!

chenrui333 commented:
@dgreatwood, thanks for your first contribution to homebrew-core! Appreciate your efforts in here. :100
